woocommerce/woocommerce-gateway-stripe
# Feature: Show UPE opt-in banner

Fixes #1693

### Description
Adding a UPE opt-in banner in the UPE payment method pages. The new banner which is created as a React component is inserted within the old UI of the pages.

### Changes proposed in this PR
- Created a `Pill` component to show the `Early Access` tag. (styles are copied from WCPay)
- Created a `UPEOptInBanner` component for the main card
- Created separate component for `all-payment-methods` icon
- Added entry in the webpack config file exposing the `UPEOptInBanner` file path
- Added `admin_options` function in `WC_Stripe_Payment_Gateway`. This is printing the form fields and inserting an empty `div` with `id=wc-stripe-upe-ont-in-banner` if `_wcstripe_feature_upe_settings` flag is enabled. The form fields are untouched otherwise.
- Added an `admin_script` to register the generated js and css files of the new `UPEOptInBanner` component in the abstract class.
- Enqueueing the script in the constructors of each UPE payment methods class.

Desktop view:
![desktop](https://user-images.githubusercontent.com/33387139/130948204-8945365b-5d7f-462a-a403-283b8aa6db4e.png)

Mobile view:
![mobile](https://user-images.githubusercontent.com/33387139/130948226-77353e19-3123-48c3-aa4d-46273fb8524e.png)

### Testing instructions
- Enable the `_wcstripe_feature_upe_settings` flag. (One way is to simply returning `true` from [this function](https://github.com/woocommerce/woocommerce-gateway-stripe/blob/e9c04deb2771f46ad082e2cde3b45f785c160ff6/includes/class-wc-stripe-feature-flags.php#L21))
- Check all the UPE payment method pages
- Make sure the QA steps are satisfied

### QA steps

- [ ] Banner is visible in all the UPE payment method pages when feature flag is on
- [ ] Banner is absent from all the UPE payment method pages when feature flag is off
- [ ] Banner is absent from all the main Stripe page
- [ ] Desktop view supports the design
- [ ] Mobile view supports the design
- [ ] UI is responsive
